internal/dataplane/kong_client.go

```diff
@@ -461,17 +462,45 @@ func (c *KongClient) sendOutToGatewayClients(
 	ctx context.Context, s *kongstate.KongState, config sendconfig.Config,
 ) ([]string, error) {
 	gatewayClients := c.clientsProvider.GatewayClients()
-	c.logger.V(util.DebugLevel).Info("sending configuration to gateway clients", "count", len(gatewayClients))
-	shas, err := iter.MapErr(gatewayClients, func(client **adminapi.Client) (string, error) {
-		return c.sendToClient(ctx, *client, s, config)
-	})
-	if err != nil {
-		return nil, err
-	}
 	previousSHAs := c.SHAs
+	gatewayClientURLs := []string{}
+	for _, cl := range gatewayClients {
+		gatewayClientURLs = append(gatewayClientURLs, cl.BaseRootURL())
+	}
 
-	sort.Strings(shas)
-	c.SHAs = shas
+	if dataplaneutil.IsDBLessMode(c.dbmode) {
```

Comment:
Please consider adding a `KongClient` unit test that would verify that in the case of a DB mode, only one client gets config updates per `Update` call (could be something similar to `TestKongClientUpdate_AllExpectedClientsAreCalledAndErrorIsPropagated`).

Reply:
Updated the implementation to call API by service, so this test to ensure all requests sent to ONE instance should NOT present,